core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -357,6 +378,14 @@ private[chisel3] object ir {
       _commandsBuilder += c
     }
 
+    def appendToPlaceholder[A](placeholder: Placeholder)(thunk: => A): A = {
+      val oldPoint = _commandsBuilder
```

Comment:
This looks a little tricksy, but is probably not likely to break presently?

I'm worried about interaction with this and use of `withRegion` where we might insert into a placeholder (because we've mutated the block to point to its buffer) instead of the intended normal block buffer.  Or anything that closes the block, so on.